usr/src/uts/common/fs/zfs/sys/zap.h

```diff
@@ -88,22 +89,12 @@ extern "C" {
 
 /*
  * Specifies matching criteria for ZAP lookups.
+ * MT_IGNORE_FORM - normalize the lookup according to FS settings.
+ * MT_PRESERVE_CASE - honor (preserve) the case in the lookup.
```

Comment:
I think that this flag is only used when MT_IGNORE_FORM is also specified, and it has the effect of doing a case-sensitive lookup even though MT_IGNORE_FORM is specified and zap_normflags includes U8_TEXTPREP_TOUPPER.  If that's right, can you expand the comment to explain that?

I think this would be better named `MT_MATCH_CASE` or `MT_CASE_SENSITIVE`.  "Preserve case" means something else: when we create an entry, we "preserve case", i.e. store it in the case specified, even though lookups can be case-insensitive.  This way, directory listings preserve the case that was used when the entry was created.